x-pack/legacy/plugins/infra/public/components/autocomplete_field/autocomplete_field.tsx

```diff
@@ -156,6 +157,12 @@ export class AutocompleteField extends React.Component<
       case 'End':
         this.updateSuggestions();
         break;
+      case 'Backspace':
+        // if the user has just deleted everything, submit to reset
+        if (this.props.value === '') {
```

Comment:
How about we move this to either the `changeValue` method or even out of this component into the `onChange` handler passed in? Then it wouldn't work for just the `Backspace` key but also for delete or any other means of emptying the input.

Reply:
Good idea. Moving to `changeValue` presented some race condition stuff where the changed state may not be updated and `this.submit()` only uses the `value` from `this.props`, but we are already checking for the existence of a new `value` prop in `componentDidUpdate` so I moved it there and verified it all still works as expected (but now it also works if you "ctrl-x" cut, for example).